src/qunit.cpp

```diff
@@ -30,7 +30,11 @@
 
 #define SHARD_STATE(shard) (norm(shard.amp0) < (ONE_R1 / 2))
 #define UNSAFE_CACHED_CLASSICAL(shard) ((norm(shard.amp0) < min_norm) || (norm(shard.amp1) < min_norm))
```

Comment:
Anytime something is named `UNSAFE` it basically requires some commentary very very nearby talking about what `UNSAFE` means in this context, and why or why not the caller should use it.

Reply:
I added a comment.